test/proxy.js

```diff
@@ -171,19 +171,13 @@ contract('Proxy', (accounts) => {
     assert.equal(web3.fromWei(balance, 'ether').toNumber(), proxyBalance, 'Balance of proxy should not have changed')
   })
 
-  it('Should throw if sending zero ether', async function() {
-    let errorThrown = false
-    let ethToSend = -2
+  it('Should not change balance if sending zero ether', async function() {
+    let ethToSend = 0
     let receiver = accounts[3]
     let proxyBalance
     let balance = await web3.eth.getBalanceAsync(proxy1.address)
     proxyBalance = web3.fromWei(balance, 'ether').toNumber()
-    try {
-      await proxy1.forward(receiver, web3.toWei(ethToSend, 'ether'), '0x0', {from: ownerProxy1})
-    } catch(e) {
-      errorThrown = true
-    }
-    assertThrown(errorThrown, 'An error should have been thrown')
+    await proxy1.forward(receiver, web3.toWei(ethToSend, 'ether'), '0x0', {from: ownerProxy1})
```

Comment:
Because of change in the nature of the test, it's not clear if the desired behaviour is to pass or fail on 0 value and 0 data.
Whichever is preferred, it might be good to actually check the tx receipt from this `await`. 

That might have to wait on resolution to the current 'low gas replay griefing attack'.

Reply:
Agreed on checking the transaction receipt.